modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/KotlinSpringServerCodegen.java

```diff
@@ -579,6 +588,11 @@ public void setReturnContainer(final String returnContainer) {
                         });
                     });
                 }
+                final List<CodegenParameter> allParams = operation.allParams;
+                if (allParams != null) {
+                    allParams.forEach(param ->
+                        param.dataType = getNonMutableContainerTypeIfNeeded(param.dataType));
+                }
```

Comment:
@rm3l could you please explain what this does and why it's needed? Thanks

Reply:
Sure. While testing, I noticed that changing the type mappings to their Mutable types resulted in all types being changed as well in the generated Spring Request handlers methods. 
This included the types for all query parameters. On the other side, the generated tests would still use the non-mutable types, resulting in the generated test code not compiling at all.

For example, given the following operation in an OpenAPI definition:
```yaml
/pet/findByStatus:
    get:
      operationId: findPetsByStatus
      parameters:
        - name: status
          in: query
          required: true
          type: array
          items:
            type: string
            enum:
              - available
              - pending
              - sold
```

Without the changes here, this resulted in the following code generated by the `kotlin-spring` generator:

- `PetApiController.kt` (notice that `status` is generated as a `MutableList`):
```kotlin
   @RequestMapping(
        method = [RequestMethod.GET],
        value = ["/pet/findByStatus"]
    )
    fun findPetsByStatus(... 
      @RequestParam(value = "status", required = true) 
      status: kotlin.collections.MutableList<kotlin.String>
    ): ResponseEntity<List<Pet>> {
        return ResponseEntity(service.findPetsByStatus(status), HttpStatus.valueOf(200))
    }
```

- `PetApiTest.kt` (which would not compile because `api.findPetsByStatus` expects a `MutableList`, but `status` is suprisingly generated as a `List`)
```kotlin
    @Test
    fun findPetsByStatusTest() {
        val status:kotlin.collections.List<kotlin.String> = TODO()
        val response: ResponseEntity<List<Pet>> = api.findPetsByStatus(status)

        // TODO: test validations
    }
```

Maybe this is not the right approach here or maybe I missed something, but I found no clear way to distinguish between the type mappings used for generating models (Kotlin data classes) and request parameters in the `kotlin-spring` generator. Also, since the original issue I reported was more related to model mutability (and not, I thought it would be ok not to change the code generated for the Spring Controllers.
Another possible approach I thought of that would fix the compilation issue was to make sure the test code generates request params as mutable containers instead.
Let me know your thoughts.

Reply:
Thanks for explaining, this way we have an explanation in case we need it 👍

Reply:
FYI, I added few comments in the code for reference.